File: Samples/Notifications/App/CppUnpackagedAppNotifications/CppUnpackagedAppNotifications/NotifyUser.cpp
#include "pch.h"
#include "NotifyUser.h"
#include "MainPage.xaml.h"
#include <winrt/Microsoft.UI.Xaml.Controls.h>

namespace winrt
{
    using namespace Microsoft::UI::Xaml::Controls;
    using namespace winrt::CppUnpackagedAppNotifications::implementation;
}

void NotifyUser::CouldNotSendToast()
{
    winrt::MainPage::Current().NotifyUser(L"Could not send toast", winrt::InfoBarSeverity::Error);
}

void NotifyUser::ToastSentSuccessfully()
{
    winrt::MainPage::Current().NotifyUser(L"Toast sent successfully!", winrt::InfoBarSeverity::Success);
}

void NotifyUser::AppLaunchedFromNotification()
{
    winrt::MainPage::Current().NotifyUser(L"App launched from notifications", winrt::InfoBarSeverity::Informational);
}

void NotifyUser::NotificationReceived()
{
    winrt::MainPage::Current().NotifyUser(L"Notification received", winrt::InfoBarSeverity::Informational);
}

void NotifyUser::UnrecognizedToastOriginator()
{
    winrt::MainPage::Current().NotifyUser(L"Unrecognized Toast Originator", winrt::InfoBarSeverity::Error);
}

File: Samples/Notifications/App/CsUnpackagedAppNotifications/CsUnpackagedAppNotifications/NotifyUser.cs

namespace CsUnpackagedAppNotifications
{
    using Microsoft.UI.Xaml.Controls;

    class NotifyUser
    {
        public static void CouldNotSendToast()
        {
            MainPage.Current.NotifyUser("Could not send toast", InfoBarSeverity.Error);
        }

        public static void ToastSentSuccessfully()
        {
            MainPage.Current.NotifyUser("Toast sent successfully!", InfoBarSeverity.Success);
        }

        public static void AppLaunchedFromNotification()
        {
            MainPage.Current.NotifyUser("App launched from notifications", InfoBarSeverity.Informational);
        }

        public static void NotificationReceived()
        {
            MainPage.Current.NotifyUser("Notification received", InfoBarSeverity.Informational);
        }

        public static void UnrecognizedToastOriginator()
        {
            MainPage.Current.NotifyUser("Unrecognized Toast Originator", InfoBarSeverity.Error);
        }
    }
}